src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```diff
@@ -0,0 +1,386 @@
+                config = connectionBuilder.getConfig();
+            }
+
+            MQQueue queue = setQueue(connectionBuilder, config);
+            int messageType = config.getMessageType();
+            MQMessage mqMessage;
+
+            switch (messageType) {
+                case 1:
+                    mqMessage = buildRequestMessageandGetResponse(config, queueMessage);
+                    break;
+                case 4:
+                    mqMessage = buildReportMessage(config, queueMessage);
+                    break;
+                default:
+                    mqMessage = buildDatagramOrReplyMessage(config, queueMessage);
+            }
+
+            if (queue == null || mqMessage == null) {
+                log.error("Error in publishing message");
+            } else {
+                log.info("queue initialized");
+                queue.put(mqMessage);
+                queue.close();
+                log.info("Message successfully placed at " + config.getQueue() + " queue and closed");
+            }
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize queue
+     */
+    MQQueue setQueue(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQQueue queue;
+        try {
+            queue = manager.accessQueue(config.getQueue(), CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating queue " + e);
+            return null;
+        }
+        return queue;
+    }
+
+    /**
+     * Create datagram or request MQMessage
+     */
+    MQMessage buildDatagramOrReplyMessage(MQConfiguration config, String queueMessage) {
+
+        MQMessage mqMessage = new MQMessage();
+        mqMessage.messageId = config.getMessageID().getBytes();
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        log.info("Correlation ID " + config.getCorrelationID());
+        log.info("Message ID " + config.getMessageID());
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```

Comment:
This should be error, log.error("Error creating Datagram/Reply mq message" + e);